torch/fx/graph.py

```diff
@@ -243,10 +250,54 @@ def format_node(n : Node) -> Optional[str]:
         for node_str in node_strs:
             if node_str:
                 s += '\n    ' + node_str
-        if self.result:
+        if hasattr(self, 'result'):
             s += f'\n    return {format_arg(self.result)}'
         return s
 
+    def lint(self, root : Optional[torch.nn.Module] = None):
+        """
+        Runs various checks on this Graph to make sure it is well-formed. In
+        particular:
+            - Checks Nodes have correct ownership (owned by this graph)
+            - Checks Nodes appear in topological order
+            - If `root` is provided, checks that `target`s exist in `root`
+        """
+
+        # Check topo order
+        def check_arg(arg : Node, n : Optional[Node] = None) -> None:
+            context_str = f' of Node \'{n}\' ' if n else ' '
+            if arg.graph is not self:
+                raise RuntimeError(f'Argument \'{arg}\'{context_str}does not belong to this Graph, '
+                                   f'but was used as an argument! If you are copying nodes from another graph, make '
+                                   f'sure to use `arg_transform` on node_copy() to remap values\n{self}')
+            if arg not in seen_values:
+                raise RuntimeError(f'Argument \'{arg}\'{context_str}was used before it has been '
+                                   f'defined! Please check that Nodes in the graph are topologically ordered\n{self}')
+
+        seen_values : Set[Node] = set()
+        for node in self.nodes:
+            if node.graph is not self:
+                raise RuntimeError(f'Node \'{node}\' does not belong to this Graph!')
+            map_arg(node.args, lambda arg: check_arg(arg, node))
+            map_arg(node.kwargs, lambda arg: check_arg(arg, node))
+            seen_values.add(node)
+
+        if hasattr(self, 'result'):
+            map_arg(self.result, check_arg)
+
+        # Check targets are legit
+        for node in self.nodes:
+            if node.op in ['get_attr', 'call_module']:
```

Comment:
do you want to check that `op` is valid too? (since it's just a string and someone could overwrite it later)